src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -694,6 +694,40 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <!--
+    ============================================================
+                                        _CreateSingleFileHost
+      Create the single-file host for the publish scenario if app is being published as a self-contained single-file .net5+ app
+    ============================================================
+     -->
+  <UsingTask TaskName="CreateAppHost" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_CreateSingleFileHost"
+          Inputs="@(IntermediateAssembly);$(SingleFileHostSourcePath)"
+          Outputs="$(SingleFileHostIntermediatePath)"
+          DependsOnTargets="_GetAppHostPaths"
+          Condition="'$(PublishSingleFile)' == 'true' and
+                     '$(SelfContained)' == 'true' and
+                     '$(SingleFileHostSourcePath)' != '' and
+                     $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0')) and
+                     Exists('@(IntermediateAssembly)') and
+                     Exists('$(SingleFileHostSourcePath)')">
+    <PropertyGroup>
+      <_UseWindowsGraphicalUserInterface Condition="($(RuntimeIdentifier.StartsWith('win')) or $(DefaultAppHostRuntimeIdentifier.StartsWith('win'))) and '$(OutputType)'=='WinExe'">true</_UseWindowsGraphicalUserInterface>
+      <_EnableMacOSCodeSign Condition="'$(_EnableMacOSCodeSign)' == '' and $([MSBuild]::IsOSPlatform(`OSX`)) and Exists('/usr/bin/codesign') and
+                                      ($(RuntimeIdentifier.StartsWith('osx')) or $(AppHostRuntimeIdentifier.StartsWith('osx')))">true</_EnableMacOSCodeSign>
```

Comment:
It looks like the logic for calculating these properties is duplicated from the [CreateAppHost task](https://github.com/dotnet/sdk/blob/c9a58063de6b6195cb7b8a19aa0c0fa89c46a58c/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L709-L711).  Would it make sense to consolidate these?

Reply:
Moved setting these to a `_GetAppHostCreationConfiguration` target